airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java

```diff
@@ -63,6 +66,50 @@ protected JsonSchemaValidator(URI baseUri) {
     this.baseUri = baseUri;
   }
 
+  /**
+   * Create and cache a schema validator for a particular schema. This validator is used when {@link #testInitializedSchema(String, JsonNode)}
+   * and {@link #ensureInitializedSchema(String, JsonNode)} is called.
+   */
+  public void initializeSchemaValidator(String schemaName, JsonNode schemaJson) {
+    schemaToValidators.put(schemaName, getSchemaValidator(schemaJson));
+  }
+
+  public boolean testInitializedSchema(String schemaName, JsonNode objectJson) {
+    final var validate = schemaToValidators.get(schemaName).validate(objectJson);
+    return validate.isEmpty();
+  }
+
+  public void ensureInitializedSchema(final String schemaName, final JsonNode objectNode) throws JsonValidationException {
+    final Set<ValidationMessage> validationMessages = schemaToValidators.get(schemaName).validate(objectNode);
+    if (validationMessages.isEmpty()) {
+      return;
+    }
+
+    throw new JsonValidationException(
+        String.format(
+            "json schema validation failed when comparing the data to the json schema. \nErrors: %s \nSchema: \n%s", Strings.join(validationMessages,
+                ", "),
+            schemaName));
+  }
+
+  /**
+   * WARNING
+   * <p>
+   * The following methods perform JSON validation **by re-creating a validator each time**. This is both CPU and GC expensive,
+   * and should be used carefully.
+   */
+
+  // todo(davin): Rewrite this section to cache schemas.
+  public boolean test(final JsonNode schemaJson, final JsonNode objectJson) {
```

Comment:
all the changes here is moving around code to make it clear what calls cache validators and what do not